### DISCUSSION TITLE
Face recognition trouble

### DISCUSSION THREAD

**Discussion by jkoopmann** [blakeblackshear/frigate#19652] [Ask A Question]
Hi,

I just started with face recognition but (as expected) get mixed results and would like to have your opinion.

Main problem: False positives. I hve tons of false positives and as stated in the documentation no way to "untrained" something. Is a picture like this suitable for training (detected face which I then learned/trained):

<img width="45" height="58" alt="image" src="https://github.com/user-attachments/assets/d574a009-79f8-497d-a0e7-2ebbb1ff1932" />

<img width="38" height="48" alt="image" src="https://github.com/user-attachments/assets/e026783d-b88a-4ab5-a0db-948b1b35ed7e" />

Or is this just to blurred/pixelled and I should refrain from using this. Unfortunately the majority of faces presented (detected) are in this quality. I have some with rather small pixel size (800) but even ones with 2000 to 5000 pixels sometimes result in this sort of blurred picture. If the general consensus is: untrained/delete all those pictures and do not use pictures less than xyz I am fine. Just need to understand it. Is the trouble coming from the fact that the first picture is a bit from the side? Or is it picture quality.

Then: If a person is detected (correctly or incorrectly) I can see them in the review section unless I filter for "person". Then those are not displayed anymore unless I reset the filter. Is this by design or an error?

Regards
  JP

**Comment by inkeep-ai-assistant** [comment]
Your face recognition issues are common during the initial training phase. Let me address your concerns based on the Frigate documentation.

**Regarding Blurry/Pixelated Training Images:**

The image quality you've shown is likely too blurry for effective training. Frigate includes a blur quality filter that adjusts confidence based on the blur level of images[(1)](https://docs.frigate.video/configuration/face_recognition). The documentation specifically recommends starting with 1-5 photos containing just the person's face that are front-facing and not turned to ensure a good starting point[(1)](https://docs.frigate.video/configuration/face_recognition).

For building a strong foundation, it's strongly recommended to only train on images that are front-facing and clear[(1)](https://docs.frigate.video/configuration/face_recognition). You should ignore images from cameras that recognize faces from an angle and aim to strike a balance between quality while having diverse conditions (day/night, different weather, different times of day)[(1)](https://docs.frigate.video/configuration/face_recognition).

The documentation advises against training on images that score 90%+ as these are already being confidently recognized[(1)](https://docs.frigate.video/configuration/face_recognition). Instead, focus on clear, lower-scoring front-facing images until the majority of front-facing images are consistently recognized correctly[(1)](https://docs.frigate.video/configuration/face_recognition).

**Addressing False Positives:**

You're correct that face recognition does not support negative training (explicitly telling it who someone is not)[(1)](https://docs.frigate.video/configuration/face_recognition). The best approach is to improve the training data by using a more diverse and representative set of images for each person[(1)](https://docs.frigate.video/configuration/face_recognition).

The documentation explains that unknown people scoring similarly to known people usually indicates the training set needs improvement and is often related to over-fitting[(1)](https://docs.frigate.video/configuration/face_recognition). This happens when you train with only a few images per person, especially if those images are very similar[(1)](https://docs.frigate.video/configuration/face_recognition).

To improve recognition:
- Review your face collections and remove most unclear or low-quality images[(1)](https://docs.frigate.video/configuration/face_recognition)
- Use the Reprocess button on each face in the Train tab to evaluate how changes affect recognition scores[(1)](https://docs.frigate.video/configuration/face_recognition)
- Focus on relatively clear images that score lower with different lighting, angles, and conditions to help the model generalize more effectively[(1)](https://docs.frigate.video/configuration/face_recognition)

**Detection Stream Quality:**

Since the majority of your detected faces are blurry, you may need to check your detect stream resolution and ensure it's sufficiently high to capture face details on person objects[(1)](https://docs.frigate.video/configuration/face_recognition). You might also need to lower your detection_threshold if faces aren't being detected properly[(1)](https://docs.frigate.video/configuration/face_recognition).

**Review Section Filter Issue:**

Regarding the review section filtering issue, this appears to be a UI behavior question that isn't directly addressed in the available documentation sources.

📚 **Sources:**
1. [Face Recognition](https://docs.frigate.video/configuration/face_recognition)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Comment by jkoopmann** [comment]
>The image quality you've shown is likely too blurry for effective training. 

Is likely or "is"? :-)

I have read the instructions. Problem: I simply do not have 4-5 photos of every person that is detected and should be detected in the future. Easy enough for family and I did this. With not real effect since the quality of the fotos is far superior than what my 12 cameras seem to detect.

Most images are front facing. What is "clear"?

Two cameras produce 1600x1200 pixels (doorbell). The others are more in the range of 2560x1440. One camera is 1920x1080 however the faces detected are pretty far away and hence are very blurry by design. I might have to simply not detect faces on them at all or increase the pixels needed dramatically. What is an easy way to determine the correct min_area setting per camera? And can I set min_area only for face_recognition as a whole or also on a per camera level?

I am using one stream per camera only for detection and recording. The stream resolution is as high as it gets...

detection_threshold is low enough since faces themselves are detected. It is false positives that is the problem.

Regarding training: If the faces I trained on (in addition to the 4-5 pictures I uploaded) are all too blurry I will not be able to use "train faces" in 95% of the cases. If the recognition display gives me 20 faces and 20 of them are blurry... How can I train?

**Reply by NickM-27** [reply]
I would not trust AIs ability to determine if the images are too blurry. The images are clear enough to make out face details which is all that matters.

**Reply by jkoopmann** [reply]
Agreed. Which is why I "reasked" hoping that someone like you chimed in. :-) So in your opinion at least the second image (front facing) is "clear enough" for training purposes? I can see the first one being misleading since the angle is a problem.

**Reply by hawkeye217** [reply]
The front facing one looks fine, yes.

**Reply by NickM-27** [reply]
For using for training it can definitely be ideal to use images with more detail. But like you said for some people that's all you can get and the face details are clear enough, so it should be fine.

**Reply by jkoopmann** [reply]
Much appreciated!

I increased min_area for the time being to focus on "better pictures" and will see that I only train on faces that are front facing and at least as good as number two. Moreover I will not add people I only have 2-3 pictures from but will maybe gather some snapshots first. That would be a feature improvement: Being able to add pictures to a person but do not enable detection so you can spend a few days/weeks to get a good footprint without running into a million false positives on the people with too few pictures first.

Once that first set is "ok" I will start to maybe decrease min_area a bit. Let's see what this gives me!

**Reply by NickM-27** [reply]
Not sure I know what you mean, there is no consequence to having a false positive recognition result

**Reply by jkoopmann** [reply]
Well currently I had > 50% false positives. Even previously correctly recognized people were suddenly misinterpreted as a new person which I just added and only had 1-2 images to work with. So I gathered (according to the documentation) to start with no less than 5 good pictures. Sometimes it takes a while to "get those". :-)

There are consequences (we can easily argue how devastating they are). 

1) People show up in stats/reviews that are not who the system thinks they are (I am allergic to false positives).
2) Once a person is "recognized" the "people filter" in review does not work anymore...

**Reply by NickM-27** [reply]
I'll have to see if I can reproduce that, realistically if other people have a strong embedding then a collection of a few images shouldn't be a better match

**Reply by jkoopmann** [reply]
Well I am at the very beginning. So even for family and closer friends I only have 5-20 pictures (and basically I am starting over right now). So for someone with 5-10 pictures I can see it happening that adding a new person with 1-2 pictures leads to what I described. 

I can only assume that once the closer family is trained on like 20-50 pictures the confidence then is strong enough for this not to happen. I would just like to make use of the time and slowly build a picture library of people occasionally coming by (once or twice a week) without giving me false positive trouble while training the strong foundation.

**Comment by mayerwin** [comment]
The "I only have blurry crops to train on" problem is exactly what pushed me to build a small companion for Frigate. It scores every crop for face quality so the blurry ones get held back automatically, and lets you label only the genuine faces, then trains those into Frigate's recognizer. It also remembers anything you mark "not a face" and auto-filters look-alikes from then on. Runs locally on CPU, only when you need it. Free and open source: https://github.com/mayerwin/frigate-better-face-recognition
